scripts/build/gen_handles.py

```diff
@@ -372,39 +372,52 @@ def main():
             inj_paths = []
             sup_paths = []
             hdls = []
+            hdls_temp = []
 
             sn = hs.node
+            hdls_temp.clear()
             if sn:
-                hdls.extend(dn.__device.dev_handle for dn in sn.__depends)
```

Comment:
If we're not sorting at the end I'd be more in favor of just replacing this line with:
`hdls.extend(sorted(dn.__device.dev_handle for dn in sn.__depends))` that way we don't need to add `.sort()` lines or have a temporary variable.

Reply:
That works for the "existing devicetree dependencies" and "devicetree support handles", as the full list is already known.  However, the "injected devicree dependencies" requires walking the list to find nodes that are in use.

```
            for inj in hs.dev_injected:
                if inj not in edt.dep_ord2node:
                    continue
                expected = edt.dep_ord2node[inj]
                if expected in used_nodes:
                    inj_paths.append(expected.path)
                    hdls.append(expected.__device.dev_handle)
```
So I think I still need a temporary list, at least for the injected dependencies.  Do you prefer I use `hdls.extend(sorted(...` for existing/support handles, and a temporary list only for the injected dependencies?